bbq/vm/vm.go

```diff
@@ -923,7 +930,10 @@ func opInvokeMethodDynamic(vm *VM, ins opcode.InstructionInvokeMethodDynamic) {
 	// Load arguments
 	arguments := vm.popN(int(ins.ArgCount))
 	receiver := arguments[ReceiverIndex]
-	arguments[ReceiverIndex] = maybeDereference(vm.context, receiver)
+	arguments[ReceiverIndex] = NewImplicitReferenceValue(
+		vm.context,
+		maybeDereference(vm.context, receiver),
+	)
```

Comment:
Similar to creating a `SelfVariable` in the interpreter, when putting the receiver in local-variables, here create an `ImplicitReferenceValue` (maybe rename to `ReceiverValue`?) and put that into locals.